main.go

```diff
@@ -0,0 +1,142 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/status-im/msgstat/stats"
+)
+
+var (
+	gitCommit    = ""      // rely on linker: -ldflags -X main.gitCommit"
+	buildStamp   = ""      // rely on linker: -ldflags -X main.buildStamp"
+	versionStamp = "0.0.1" // rely on linker -ldflags -X main.versionStamp"
+)
+
+var (
+	sourceFile = flag.String("file", "", "input file containing status-go logs")
+	targetFile = flag.String("out", "", "output file to save processed data into")
+	format     = flag.String("format", "json", "format of output data. -format=json|yaml|csv")
+	version    = flag.Bool("v", false, "Print version")
+)
+
+type wopCloser struct {
+	io.Writer
+}
+
+// Close does nothing.
+func (wopCloser) Close() error {
+	return nil
+}
+
+func main() {
+	flag.Usage = printUsage
+	flag.Parse()
+
+	// if we are to print version.
+	if *version {
+		printVersion()
+		return
+	}
+
+	if *sourceFile == "" && !hasIncomingData() {
+		printUsage()
+		return
+	}
+
+	var wc io.WriteCloser
+
+	if *targetFile != "" {
+		targetOutputFile, err := os.Create(*targetFile)
+		if err != nil {
+			log.Fatalf("Failed to open output file %+q: %v", *targetFile, err)
+			return
+		}
+
+		wc = targetOutputFile
+	} else {
+		wc = wopCloser{Writer: os.Stdout}
+	}
+
+	var rc io.ReadCloser
+
+	if *sourceFile != "" {
+		targetInputFile, err := os.OpenFile(*sourceFile, os.O_RDONLY, 0700)
+		if err != nil {
+			log.Fatalf("Failed to open input file %+q: %v", *sourceFile, err)
+			return
+		}
+
+		rc = targetInputFile
+	} else {
+		rc = ioutil.NopCloser(os.Stdin)
+	}
+
+	if err := stats.ReadAggregates(rc, wc, *format); err != nil {
+		log.Fatalf("Failed to process logs successfully: %v", err)
+		return
+	}
+
+	log.Println("Aggregation Complete!")
+
+	if *targetFile != "" {
+		log.Printf("See output in %+q\n", *targetFile)
+	}
+}
+
+func printVersion() {
```

Comment:
Surely you cannot tell a lot interesting here, but I like comments even on private functions, types, variables, etc. Mostly because they are a kind of optical separator and also sometime help to reflect if the chosen name is good (here it is).